### DISCUSSION TITLE
Mutable default argument found in jsonable_encoder

### DISCUSSION THREAD

**Discussion by joeblackwaslike** [fastapi/fastapi#8643] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
N/A
```

### Description

I discovered a bug in the jsonable_encoder function. That function contains a mutable default argument "custom_encoder", which in this case is a dictionary.

I believe the implications of this are pretty obvious, but for anyone reading along that's unfamiliar with why this is problematic and would like to learn more, I refer you to https://florimond.dev/en/posts/2018/08/python-mutable-defaults-are-the-source-of-all-evil/.

PS: I'm a huge fan of FastAPI, please keep up the good work and honored to have helped make it 1% better than it already was :)

I've already submitted a PR addressing the issue: https://github.com/tiangolo/fastapi/pull/4010

### Operating System

macOS

### FastAPI Version

HEAD

### Python Version

3.9

**Comment by YuriiMotov** [comment, marked as answer]
https://github.com/fastapi/fastapi/pull/4010#issuecomment-952208974
